src/ansys/dpf/core/field_base.py

```diff
@@ -504,6 +504,8 @@ def data_as_list(self):
 
     @data.setter
     def data(self, data):
+        if isinstance(data, (np.ndarray, np.generic)):
```

Comment:
@moe-ad would it be possible to minimize the cases in which we copy the data?
The point of doing DPF `InProcess` for example would be to NOT have to copy the data in most cases.
In case of an array view, it can be argued that the copy is required, but only in this case.